honeybadgermpc/polynomial.py

```diff
@@ -277,6 +277,124 @@ def fft(poly, omega, n, seed=None):
     return fft_helper(padded_coeffs, omega, poly.field)
 
 
+def fnt_decode_step1(poly, zs, omega2, n):
+    """
+    This needs to be run once for decoding a batch of secret shares
+    It depends only on the x values (the points the polynomial is
+    evaluated at, i.e. the IDs of the parties contributing shares) so
+    it can be reused for multiple batches.
+    Complexity: O(n^2)
+
+    args:
+        zs is a subset of [0,n)
+        omega2 is a (2*n)th root of unity
+
+    returns:
+        A(X) evaluated at 1...omega2**(2n-1)
+        Ai(xi) for each xi = omega**(zi)
+
+    where:
+        omega = omega2**2
+        where A(X) = prod( X - xj ) for each xj
+        Ai(xi) = prod( xi - xj ) for j != i
+    """
+    k = len(zs)
+    omega = omega2**2
+    xs = [omega**z for z in zs]
+
+    # Compute A(X)
+    a_ = poly([1])
+    for i in range(k):
+        print(i, poly([-xs[i], 1]))
```

Comment:
Please avoid using print statements anywhere. Use the logging framework instead. `logging.info` or `logging.debug` or `logging.warn`. Please take a pass over your code and get rid of all of them.

Reply:
Sure